src/zjs_console.c

```diff
@@ -152,18 +160,167 @@ static jerry_value_t console_error(const jerry_value_t function_obj,
     return do_print(function_obj, this, argv, argc, stderr);
 }
 
+static jerry_value_t console_time(const jerry_value_t function_obj,
+                                  const jerry_value_t this,
+                                  const jerry_value_t argv[],
+                                  const jerry_length_t argc)
+{
+    if (!jerry_value_is_string(argv[0])) {
+        ERR_PRINT("invalid parameters\n");
+        return ZJS_UNDEFINED;
+    }
+    int sz = jerry_get_string_size(argv[0]);
+    char label[sz + 1];
+    int len = jerry_string_to_char_buffer(argv[0], (jerry_char_t *)label, sz);
+    if (sz != len) {
+        ERR_PRINT("size mismatch\n");
+        return ZJS_UNDEFINED;
+    }
+    label[len] = '\0';
+
+    jerry_value_t new_timer_obj = jerry_create_object();
+    zjs_port_timer_t* timer = zjs_malloc(sizeof(zjs_port_timer_t));
+    if (!timer) {
+        ERR_PRINT("could not allocate timer object\n");
+        return ZJS_UNDEFINED;
+    }
+    zjs_port_timer_init(timer);
+    // set timer to go off every millisecond (this is the best accuracy we can get)
+    zjs_port_timer_start(timer, 1);
```

Comment:
We shouldn't be making a timer here. Just record the start time in a structure associated with the label, and calculate the end time later by subtracting this from the then-current time. That's how the performance module is intended to be used, so let's also consider whether we really need both? Please review #325 to see how it relates to this. Offhand I don't see console.time() mentioned so maybe that's a good model for us that @pfalcon didn't consider. Let's make sure he comments before we go forward with this.